File: backend/src/EasyAgree.Application/Documents/RequiredDocument.cs
using EasyAgree.Domain.Enums;

namespace EasyAgree.Application.Documents;

/// <summary>
/// One document type it's worth suggesting the user upload for a given
/// template - never an identity document (passport/PINFL/ID) since those
/// come from the account profile / MyID, never from a scan.
/// </summary>
public sealed record RequiredDocument(
    DocumentType Type,
    string Title,
    string Description,
    bool Required,
    int Priority);

File: app/lib/features/documents/domain/required_document.dart
class RequiredDocument {
  const RequiredDocument({
    required this.type,
    required this.title,
    required this.description,
    required this.required,
    required this.priority,
  });

  final String type;
  final String title;
  final String description;
  final bool required;
  final int priority;

  factory RequiredDocument.fromJson(Map<String, dynamic> json) => RequiredDocument(
    type: json['type'] as String,
    title: json['title'] as String,
    description: json['description'] as String,
    required: json['required'] as bool,
    priority: json['priority'] as int,
  );
}
